Completed Issue #2 | User Can View Products

ejgilleland opened the pull request:
![image](https://user-images.githubusercontent.com/73370554/104898482-8b938280-592e-11eb-8358-27c0f2a38fdf.png)

ejgilleland committed:
Completed Issue #2 | User Can View Products

shadorki commented on index.html:
```diff
@@ -11,13 +11,13 @@
 <body>
   <div class="header">
-    <img id="store-logo" src="images/logo.png">
+    <a href="index.html"><img id="store-logo" src="images/logo.png"></a>
```
This is going to give you an issue later on, You are going to be dynamically creating what the user sees based off which page is selected. Don't worry about it for now.

shadorki commented on main.js:
```diff
@@ -0,0 +1,39 @@
+var $productsContainer = document.querySelector('.products-container');
+var $product = document.createElement('div');
+var $productImage = document.createElement('img');
+var $productInformation = document.createElement('div');
+var $productName = document.createElement('div');
+var $icons = document.createElement('div');
+var $heartIcon = document.createElement('i');
+var $cartPlus = document.createElement('i');
```
I totally get the logic when it comes to creating a node and then cloning it for your purposes. But this hurts the level of abstraction you are trying to give to yourself.  Think of it this way. Check next comment

shadorki commented on main.js:
```diff
@@ -0,0 +1,39 @@
+var $heartIcon = document.createElement('i');
+var $cartPlus = document.createElement('i');
+
+function render() {
+  for(var i = 0; i < data.products.length; i++) {
+    $product.className = 'product';
+    $productImage.className = 'product-image';
+    $productInformation.className = 'product-information';
+    $productName.className = 'product-name';
+    $icons.className = 'icons';
+    $heartIcon.className = 'fa fa-heart';
+    $heartIcon.setAttribute('aria-hidden', 'true');
+    $cartPlus.className = 'fa fa-cart-plus';
+    $cartPlus.setAttribute('aria-hidden', 'true');
+    $productImage.setAttribute('src', data.products[i].imageUrl);
+    $productImage.setAttribute('alt', data.products[i].name);
+    $productsContainer.appendChild($product);
+    $product.append($productImage, $productInformation);
+    $productInformation.append($productName, $icons);
+    $productName.innerHTML = data.products[i].name + '<br> $' + data.products[i].price;
+    $icons.append($heartIcon, ' ', $cartPlus);
+
+    $product = $product.cloneNode();
+    $productImage = $productImage.cloneNode();
+    $productInformation = $productInformation.cloneNode();
+    $productName = $productName.cloneNode();
+    $icons = $icons.cloneNode();
+    $heartIcon = $heartIcon.cloneNode();
+    $cartPlus = $cartPlus.cloneNode();
+  }
+}
```
Define a function named createProduct
It will take an object as an argument (The object from the array in the data model)
And it will return a Dom Element.

This way you will not need to have a bunch global variables containing empty dom elements.

In your render function you can iterate through the data model, call createProduct for every object in the data model, creates a dom element, and appends them to the dom.

ejgilleland committed:
Changes made, mostly variable scoping for creation of dom elements

ejgilleland commented:
Removed the <a> element on the store logo in the header (I wasn't sure if we were gonna change the view dynamically, but that does seem like more fun!)

Moved the transition properties onto class selectors for fa-heart, fa-cart-plus, and fa-shopping-cart (previously the transition property was on the :hover pseudoclass).

Updated JS per your instructions:
1) DOM elements created via a createProduct() function that takes the product object as an argument and returns a DOM element
2) render() function iterates through each product object in data.js and calls the createProduct() function for each object, which is then appended to the parent container in the page.
